File: 0011-container-with-most-water/0011-container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int n = height.size();
        int MaxWater = 0;

        int i = 0;
        int j = n-1;

        while(i<j){
            int width = j-i;
            int h = min(height[i],height[j]);
            int area = h * width;

            MaxWater = max(MaxWater , area);

            if(height[i] > height[j]){
                j--;
            }else{
                i++;
            }


        }
        return MaxWater;

    }

};

File: 0011-container-with-most-water/0011-container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int n = height.length;
        int mostWater = 0;
        int a = 0;
        int b = n-1;
        while(a < b){
            int Height = Math.min(height[a],height[b]);
            int width = b - a;
            int water = Height * width;
            mostWater = Math.max(mostWater,water);
            if(height[a] <= height[b]){
                a++;
            }else{
                b--;
            }
        }
        return mostWater;
        
    }
}